### PR TITLE
Consul API Gateway, Add Tolerations support

### PR DESCRIPTION
**Pull request by sarahalsmiller**
Changes proposed in this PR:
- At request of customer, add support for configuring tolerations to consul api gateway controller and gateway pods

How I've tested this PR:
- Updated Conformance test matrix in https://github.com/hashicorp/consul-api-gateway/pull/426 to run the branch referenced in this PR

How I expect reviewers to test this PR:
- Tests in https://github.com/hashicorp/consul-api-gateway/pull/426 pass
- Tests on this PR pass

Checklist:
- [ ] Tests added
- [ X ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/templates/api-gateway-controller-deployment.yaml [hunk 1]
```diff
@@ -237,4 +237,8 @@ spec:
       nodeSelector:
         {{ tpl .Values.apiGateway.controller.nodeSelector . | indent 8 | trim }}
       {{- end }}
+      { { - if .Values.apiGateway.controller.tolerations } }
```
Comment (thisisnotashwin) [inline charts/consul/templates/api-gateway-controller-deployment.yaml:240]: The formatting on the `{{` and `}}` needs correcting here. There is an additional space.

File: charts/consul/templates/api-gateway-gatewayclassconfig.yaml [hunk 2]
```diff
@@ -35,10 +35,14 @@ spec:
   image:
     consulAPIGateway: {{ .Values.apiGateway.image }}
     envoy: {{ .Values.global.imageEnvoy }}
-  {{- if .Values.apiGateway.nodeSelector }}
+  {{- if .Values.apiGateway.managedGatewayClass.nodeSelector }}
   nodeSelector:
     {{ tpl .Values.apiGateway.managedGatewayClass.nodeSelector . | indent 4 | trim }}
   {{- end }}
+  { { - if .Values.apiGateway.managedGatewayClass.tolerations } }
```
Comment (thisisnotashwin) [inline charts/consul/templates/api-gateway-gatewayclassconfig.yaml:42]: Same thing with the `{{` and `}}` here as well.

Comment (sarahalsmiller) [inline charts/consul/templates/api-gateway-gatewayclassconfig.yaml:42]:
```suggestion
  {{ - if .Values.apiGateway.managedGatewayClass.tolerations }}
```

File: charts/consul/templates/api-gateway-controller-deployment.yaml (hunk 1 above)
Comment (sarahalsmiller) [inline charts/consul/templates/api-gateway-controller-deployment.yaml:240]:
```suggestion
      {{ - if .Values.apiGateway.controller.tolerations }}
```

**Comment by t-eckert** [comment]
Just a drive-by comment. I see some failures in the BATS tests, is this something we want to get passing before merging? https://github.com/hashicorp/consul-k8s/actions/runs/3362576972/jobs/5574777333

**Comment by sarahalsmiller** [comment]
> https://github.com/hashicorp/consul-k8s/actions/runs/3362576972/jobs/5574777333



> Just a drive-by comment. I see some failures in the BATS tests, is this something we want to get passing before merging? https://github.com/hashicorp/consul-k8s/actions/runs/3362576972/jobs/5574777333



> Just a drive-by comment. I see some failures in the BATS tests, is this something we want to get passing before merging? https://github.com/hashicorp/consul-k8s/actions/runs/3362576972/jobs/5574777333

I'm assuming so, but I'm not entirely sure where those failures are coming from at the moment

**Comment by t-eckert** [comment]
> I'm assuming so, but I'm not entirely sure where those failures are coming from at the moment

Sure! We have a suite of unit tests that run with BATS. These perform the Helm template action that creates the Kubernetes resources from the Helm values, then validates their structure.

The [contributing guide](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#writing-unit-tests) has more information that may be helpful.

**Comment by t-eckert** [comment]
@sarahalsmiller, looks like those Helm tests are no longer failing! 🥳

File: charts/consul/values.yaml [hunk 3]
```diff
@@ -2897,6 +2897,13 @@ apiGateway:
     # @type: string
     nodeSelector: null
 
+    # This value defines the tolerations that will be assigned to a gateway pod
```
Comment (jmurret) [inline charts/consul/values.yaml:2900]:
```suggestion
    # This value defines the tolerations that will be assigned to a gateway pod.
```

**Review by jmurret — APPROVED** [review]
👏 nice!  one small punctuation suggestion, but not blocking.

**Pull request merged**